carmajava/signal_plugin/src/main/java/gov/dot/fhwa/saxton/carma/signal_plugin/TrafficSignalPlugin.java

```diff
@@ -390,15 +411,103 @@ static protected SpatMessage convertSpatMessage(IntersectionData data) {
+                replanning.set(false);
+                return;
+            }
+
+            // Generate plan in new thread
+            Thread planningThread = new Thread(new Runnable() {
+
+                @Override
+                public void run() {
+                    log.info("Attempting Ead Plan");
+                    // Planning
+                    double currentDowntrack = pluginServiceLocator.getRouteService().getCurrentDowntrackDistance();
+                    double speedLimit = pluginServiceLocator.getRouteService().getSpeedLimitAtLocation(currentDowntrack).getLimit();
+                    DataElementHolder state = null;
+                    List<Node> eadResult = null;
+                    
+                    for (int i = 0; i < 3; i++) {
+                        state = getCurrentStateData(speedLimit);
+        
+                        eadResult = generatePlan(state);
+                        if (eadResult != null) {
+                            break;
+                        }
+                    }
+                    
+                    // Store the current plan
+                    DoubleDataElement startTime = (DoubleDataElement) state.get(DataElementKey.PLANNING_START_TIME);
+                    DoubleDataElement startDowntrack = (DoubleDataElement) state.get(DataElementKey.PLANNING_START_DOWNTRACK);
+        
+                    setPlan(eadResult, startDowntrack.value());
+                    
+                    if (eadResult == null) {
+                        log.warn("Ead result was null requesting plan with no availability");
+                        setAvailability(false);
+                        pluginServiceLocator.getArbitratorService().requestNewPlan();
+                        replanning.set(false);
+                        return;
+                    }
+        
+                    log.info("EadAStar result is path of size: " + eadResult.size());
+                    log.info("EadAStar Num ANA iterations: " + ANAStarSolver.iterationCount);
+                    // Set the new plan as the current plan for collision checker
+                    collisionChecker.setHostPlan(eadResult, startTime.value(), startDowntrack.value());
+        
+                    log.info("Planning Successful Requesting replan with availability: " + availability);
+                    setAvailability(availability);
+                    pluginServiceLocator.getArbitratorService().requestNewPlan();
+                }
+
+            });
+
+            planningThread.start();
+
+            
+        }
+    }
+
+    private List<Node> generatePlan(DataElementHolder state) {
+        // Try 3 times to plan. With updated state data each time if needed. We will accept the first completed plan
+        log.info("Requesting AStar plan with intersections: " + intersections.toString());
+
+        try {
+            // TODO remove this print after NCV handling is stable
```

Comment:
Maybe just put this at `DEBUG` level, so if we need it we have it but it won't be output most of the time

Reply:
This log is extremely useful for testing NCVs, but activating debug logs for guidance slows everything down too much. I really wish I could control debug level on a per-class basis. Maybe there is a way but I don't know how. I would rather leave this for now.